src/lib/journey/stages/generic.svelte

```diff
@@ -77,7 +78,9 @@
     }
   });
 
-  onMount(() => captureLinks(linkWrapper, journey));
+  onMount(() => {
+    return Promise.all([captureLinks(linkWrapper, journey), initializeProtect()]);
```

Comment:
Are we intending to initialize Protect at the start of a journey regardless of condition? Plus, we're providing a callback component to handle additional initialization. I'm a bit confused by this.

Reply:
Well - yes this function would be invoked, but no pingProtect is not necessarily initialized or attempted to be without the journey config set in the start options

Reply:
I'm still not sure I understand. Essentially, we want two approaches to initializing/starting the Protect feature in the Widget:

1. Only when the Widget encounters the callback in a Journey
2. When the dev explicitly initializes using an external method call from the Widget

I don't think we should implicitly initialize it, like you have here. I may be missing something, but let's chat about it.

Reply:
I guess what i'm saying is the function has a check to see if the code was initialized. so yes the function is invoked, but if the setting is not in the journeyStart options, the code never makes any calls it just passes through

https://github.com/ForgeRock/forgerock-web-login-framework/pull/327/files#diff-07b34af429841bd1a5bb3b910d439b7adcb4ec8d79d5ade2ab7c199bb06eb4e0R14

Is this not sufficient?

Reply:
this no longer applies